src/Microsoft.AspNetCore.Mvc.TagHelpers/InputTagHelper.cs

```diff
@@ -189,20 +189,21 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
             TagBuilder tagBuilder;
             switch (inputType)
             {
-                case "checkbox":
-                    GenerateCheckBox(modelExplorer, output);
-                    return;
-
```

Comment:
Nope, in the https://github.com/aspnet/Mvc/pull/5109/commits/4b297d9bca77eea3ef526a65535f1d16835fb5d8 commit, I was removed `case "hidden"`, when I added it again I added it on top. (I thought `hidden`s are more common then `checkbox`es.)